components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue

```diff
@@ -1,50 +1,189 @@
+import { JustInteraction, createInteraction } from '@kodadot1/minimark'
+import nftByIdMinimal from '@/queries/rmrk/subsquid/nftByIdMinimal.graphql'
+import useRmrkVersion from '@/composables/useRmrkVersion'
+const props = withDefaults(
+  defineProps<{
+    nftId: string
+    currentOwner?: string
+    collectionId?: string
+    nftPrice?: string
+  }>(),
+  {
+    nftId: '',
+    currentOwner: '',
+    collectionId: '',
+    nftPrice: '',
+  }
+)
+
+const { urlPrefix, client } = usePrefix()
+const { accountId } = useAuth()
+const { $store, $apollo, $i18n } = useNuxtApp()
+const { apiInstance } = useApi()
+const emit = defineEmits(['buyNFT'])
+const ACTION = 'BUY'
+const actionLabel = $i18n.t('nft.action.buy')
 
-const { urlPrefix } = usePrefix()
-const { nft } = useGalleryItem()
+const { howAboutToExecute, initTransactionLoader, isLoading, status } =
+  useMetaTransaction()
 
 const active = ref(false)
-const label = computed(() => (active.value ? 'Confirm' : 'Buy'))
+const label = computed(() =>
+  active.value ? $i18n.t('nft.action.confirm') : $i18n.t('nft.action.buy')
+)
+
+const balance = computed<string>(() => {
+  if (urlPrefix.value == 'rmrk') {
+    return $store.getters.getAuthBalance
+  }
+  return $store.getters.getTokenBalanceOf(getKusamaAssetId(urlPrefix.value))
+})
+const disabled = computed(() => {
+  if (!(props.nftPrice && balance.value)) {
+    return false
+  }
+  return Number(balance.value) <= Number(props.nftPrice)
+})
+
+function onClick() {
+  if (active.value) {
+    handleBuy()
+  } else {
+    active.value = true
+  }
+}
+// close the buy button when transaction loading is finsihed
+watch(isLoading, (loading) => {
+  active.value = loading
+})
+
+const getTranasactionParams = async () => {
+  const api = await apiInstance.value
+  if (urlPrefix.value == 'rmrk') {
+    const rmrk = createInteraction(
+      ACTION as JustInteraction,
+      useRmrkVersion().version,
+      props.nftId,
+      ''
+    )
+    return {
+      cb: api.tx.utility.batchAll,
+      arg: [
+        [
+          api.tx.system.remark(rmrk),
+          api.tx.balances.transfer(props.currentOwner, props.nftPrice),
+          somePercentFromTX(api, props.nftPrice),
+        ],
+      ],
+    }
+  }
 
-function toggleActive() {
-  active.value = !active.value
+  // not RMRK
+  const { id: collectionId, item: itemId } = tokenIdToRoute(props.nftId)
+  return {
+    cb: getApiCall(api, urlPrefix.value, ACTION),
+    arg: [collectionId, itemId],
+  }
+}
+const checkBuyBeforeSubmit = async () => {
+  const nft = await $apollo.query({
+    query: nftByIdMinimal,
+    client: client.value,
+    variables: {
+      id: props.nftId,
+    },
+  })
+
+  const {
+    data: { nft: nFTEntity },
+  } = nft
+
+  if (
+    nFTEntity.currentOwner !== props.currentOwner ||
+    nFTEntity.burned ||
+    nFTEntity.price === 0 ||
+    nFTEntity.price !== props.nftPrice
+  ) {
+    showNotification(
+      $i18n.t('nft.notification.nftChanged', {
+        chain: 'RMRK',
```

Comment:
use urlPrefix here?

Reply:
nvm this feature is only used on rmrk, we might want to have it on other chain

Reply:
any change required here? resolve?

Reply:
yup please use URL prefix

Reply:
done